### PR TITLE
[IO] Take TBufferMerger out of the Experimental namespace

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

**Review by pcanal — APPROVED** [review]

File: io/io/inc/ROOT/TBufferMerger.hxx [hunk 1]
```diff
@@ -213,7 +213,14 @@ public:
...
 };
 
+namespace Experimental {
+using TBufferMerger R__DEPRECATED(
+   6, 28, "Please use ROOT::TBufferMerger instead of ROOT::Experimental::TBufferMerger.") = ::ROOT::TBufferMerger;
+using TBufferMergerFile
+   R__DEPRECATED(6, 28, "Please use ROOT::TBufferMergerFile instead of ROOT::Experimental::TBufferMergerFile.") =
+      ::ROOT::TBufferMergerFile;
 } // namespace Experimental
+
```
Comment (amadio) [inline io/io/inc/ROOT/TBufferMerger.hxx:224]: Is this really enough to ensure all current code still compiles without problems? I'd ask CMS to test first.

Comment (eguiraud) [reply io/io/inc/ROOT/TBufferMerger.hxx:224]: how do I ask CMS to test a PR? but also, why shouldn't this be enough (they'll have deprecation warnings, but ROOT::Experimental::TBufferMerger is there)?

**Comment by eguiraud** [comment]
What do we want to do with the LinkDef entries? Only ROOT::TBufferMerger or also ROOT::Experimental::TBufferMerger? probably both until Experimental::TBufferMerger is removed?

**Comment by Axel-Naumann** [comment]
We could ask our good friends from CMS: @davidlange6 or @smuzaffar - could you cross-check that this change doesn't break CMSSW? If you prefer you can also just tell us to go ahead and merge, and we'll deal with breaking CMSSW in a subsequent step - in this case it's just a heads-up :-) Let us know your preference, please. (Thanks @amadio for pointing this out!)

**Comment by smuzaffar** [comment]
@Axel-Naumann , I have started a PR test here https://github.com/cms-sw/root/pull/157 . It will take few hours before we have the results.

**Comment by Axel-Naumann** [comment]
@eguiraud LinkDef just needs `ROOT::TBufferMerger` - ROOT I/O knows the other one is just a typedef.

But we don't really support I/O *of* `TBufferMerger`, so we just need to provide the `ClassDef` functions' implementation in the dictionary.

**Comment by eguiraud** [comment]
@Axel-Naumann ok, I removed the `Experimental::` spelling from the LinkDef

**Comment by eguiraud** [comment]
@smuzaffar please let me know if/when it's ok to merge this (although I also checked and there is no TBufferMerger usage in CMSSW).

**Comment by smuzaffar** [comment]
@eguiraud , cmssw test passed, so feel free to integrate it.

**Pull request merged**